lib/presentation/reset_password/reset_password_view.dart

```diff
@@ -0,0 +1,218 @@
+import 'package:appetizer/data/core/theme/dimensional/dimensional.dart';
+import 'package:appetizer/presentation/components/loading_indicator.dart';
+import 'package:appetizer/presentation/login/components/login_button.dart';
+import 'package:appetizer/presentation/reset_password/bloc/reset_password_bloc.dart';
+import 'package:appetizer/presentation/reset_password/components/ResetPasswordTile/reset_password_banner.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+@RoutePage()
+class ResetPasswordScreen extends StatelessWidget {
+  final TextEditingController _oldPasswordController = TextEditingController();
+  final TextEditingController _newPasswordController = TextEditingController();
+  final TextEditingController _confirmPasswordController =
+      TextEditingController();
+  ResetPasswordScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => ResetPasswordBloc(userRepository: context.read()),
+      child: Scaffold(
+        backgroundColor: Colors.white,
+        body: BlocConsumer<ResetPasswordBloc, ResetPasswordState>(
+          listener: (context, state) {
+            if (state is ResetPassword && state.error != null) {
+              ScaffoldMessenger.of(context).showSnackBar(
+                SnackBar(
+                  content: Text(state.error!),
+                  backgroundColor: Colors.red,
+                ),
+              );
+            }
+            if (state is ResetPasswordSuccess) {
+              ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+                content: Text('Password reset successfully!'),
+                backgroundColor: Colors.green,
+              ));
+              context.router.pop();
+            }
+          },
+          builder: (context, state) {
+            if (state is ResetPassword) {
+              return Column(
+                children: [
+                  const ResetPasswordBanner(),
+                  Expanded(
+                    child: SingleChildScrollView(
+                      padding: 24.toHorizontalPadding,
+                      child: Column(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Text(
+                            'Enter your old password',
+                            style: GoogleFonts.notoSans(
+                              fontSize: 18,
+                              fontWeight: FontWeight.w600,
+                            ),
+                          ),
+                          20.toVerticalSizedBox,
+                          TextField(
```

Comment:
@am-casper TextField section seems quite repetitive. Abstract this part into a different widget like AppTextField and use it.

Reply:
Resolved.